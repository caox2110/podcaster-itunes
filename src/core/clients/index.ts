export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
