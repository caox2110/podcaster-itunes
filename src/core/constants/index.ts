import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
