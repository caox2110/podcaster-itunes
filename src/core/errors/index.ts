export { default as ExceptionHandler } from './Exceptions.handler';
