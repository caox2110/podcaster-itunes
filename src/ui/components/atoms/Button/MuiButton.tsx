import Button from '@mui/material/Button';
import type { ButtonProps } from '@mui/material';
import { ReactNode } from 'react';

/**
 * Type of MuiButton component props.
 * ItÂ´s can receive the ButtonProps and React.ButtonHTMLAttributes<HTMLButtonElement>.
 *
 * @typedef {ButtonProperties}
 */
type ButtonProperties = {
  children: ReactNode;
  'data-testid'?: string;
} & ButtonProps &
  React.ButtonHTMLAttributes<HTMLButtonElement>;

/**
 * MuiButton Component definition. An button html element.
 *
 * @param {ButtonProperties} { children, 'data-testid': dataTestId, ...rest }
 * @returns
 */
function MuiButton({ children, 'data-testid': dataTestId, ...rest }: ButtonProperties) {
  return (
    <Button data-testid={dataTestId} {...rest}>
      {children}
    </Button>
  );
}

/**
 * Default values for MuiButton component props
 *
 * @type {{ 'data-testid': string; color: string; variant: string; }}
 */
const defaultProps = {
  'data-testid': '',
  color: 'primary',
  variant: 'outlined',
};

MuiButton.defaultProps = defaultProps;

export type { ButtonProperties };
export default MuiButton;
