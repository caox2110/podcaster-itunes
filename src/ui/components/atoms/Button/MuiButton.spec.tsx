import { fireEvent, render } from '@/core/test';
import type { RenderResult } from '@/core/test';

import MuiButton, { ButtonProperties } from './MuiButton';

type RenderAppType = {
  button: HTMLButtonElement;
} & RenderResult;

const properties = {
  'data-testid': 'home-podcast-button',
  children: 'Podcaster',
};

function renderApp({ ...rest }: ButtonProperties = properties): RenderAppType {
  const utils = render(<MuiButton {...rest} />);
  const { getByDataTestId } = utils;

  const button = getByDataTestId(/home-podcast-button/) as HTMLButtonElement;

  return {
    ...utils,
    button,
  };
}

describe('MuiButton', () => {
  describe('Rendering', () => {
    it('Should render without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });

  describe('Appearance', function () {
    it('should render outlined style by default', () => {
      const { button } = renderApp();
      expect(button.className).toMatch(/outlined/i);
    });
  });

  describe('Default behaviors', function () {
    it('should call onClick event', async function () {
      const handleClick = jest.fn();
      const { button } = renderApp({ ...properties, onClick: handleClick });
      fireEvent.click(button);
      expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });
});
