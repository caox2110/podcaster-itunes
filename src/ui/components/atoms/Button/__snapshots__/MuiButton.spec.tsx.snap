// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiButton Rendering Should render without throwing any errors 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1no0c5y-MuiButtonBase-root-MuiButton-root"
    data-testid="home-podcast-button"
    tabindex="0"
    type="button"
  >
    Podcaster
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
